sharedbuffers/mapped_struct.py

```diff
@@ -1976,8 +1975,27 @@ def _intersect_2(self, seq):
         elif not seq:
             return proxied_frozenset(None, 0, 0)
 
-        rv = self._add_to(set())
-        rv.intersection_update(seq)
+        if len(self) > len(seq):
+            rv = set(seq)
+            typ = type(seq)
+
+            if typ is set:
+                set_seq = cython.cast(set, seq)
```

Comment:
cython.cast is weird in pure python, just declare the type of set_seq in locals and leave the cast implicit